khelo/lib/ui/flow/tournament/detail/tabs/tournament_detail_overview_tab.dart

```diff
@@ -0,0 +1,416 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:khelo/components/won_by_message_text.dart';
+import 'package:khelo/domain/extensions/context_extensions.dart';
+import 'package:khelo/domain/extensions/enum_extensions.dart';
+import 'package:khelo/domain/extensions/string_extensions.dart';
+import 'package:khelo/domain/formatter/date_formatter.dart';
+import 'package:khelo/ui/app_route.dart';
+import 'package:style/animations/on_tap_scale.dart';
+import 'package:style/extensions/context_extensions.dart';
+import 'package:style/text/app_text_style.dart';
+
+import '../../../../../components/image_avatar.dart';
+
+class TournamentDetailOverviewTab extends ConsumerStatefulWidget {
+  final TournamentModel tournament;
+
+  const TournamentDetailOverviewTab({
+    super.key,
+    required this.tournament,
+  });
+
+  @override
+  ConsumerState<TournamentDetailOverviewTab> createState() =>
+      _TournamentDetailOverviewTabState();
+}
+
+class _TournamentDetailOverviewTabState
+    extends ConsumerState<TournamentDetailOverviewTab> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      color: context.colorScheme.containerLow,
+      child: ListView(
+        padding: EdgeInsets.symmetric(horizontal: 16)
+            .copyWith(bottom: context.mediaQueryPadding.bottom + 40),
+        children: [
+          _featuredMatchesView(context, widget.tournament.matches),
+          _keyStatsView(context, widget.tournament.keyStats),
+          _teamsSquadsView(context, widget.tournament.teams),
+          _infoView(context, widget.tournament),
+        ],
+      ),
+    );
+  }
+
+  Widget _featuredMatchesView(BuildContext context, List<MatchModel> matches) {
+    if (matches.isEmpty) return SizedBox();
+
+    return Padding(
+      padding: const EdgeInsets.only(top: 16),
+      child: Column(
+        children: [
+          _header(
+            context,
+            title:
+                context.l10n.tournament_detail_overview_featured_matches_title,
+            showViewAll: matches.length > 3,
+            onViewAll: () {},
+          ),
+          ...List.generate(
+            matches.take(3).length,
+            (index) => _matchCellView(context, matches[index]),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget _matchCellView(BuildContext context, MatchModel match) {
+    return Container(
+      padding: EdgeInsets.symmetric(horizontal: 16, vertical: 24),
+      margin: EdgeInsets.symmetric(vertical: 8),
+      decoration: BoxDecoration(
+        color: context.colorScheme.surface,
+        borderRadius: BorderRadius.circular(16),
+      ),
+      child: Row(
+        children: [
+          _buildTeamInfo(team: match.teams.first.team),
+          const Spacer(),
+          Column(
+            children: [
+              if (match.matchResult != null) ...[
+                WonByMessageText(
```

Comment:
since it's one element, in `if` can we move it outside of column and apply condition there.